### PR TITLE
[NetAppFiles] Anf 27235 update for 2022-11-01-preview

### PR DESCRIPTION
**Pull request by audunn**
# ARM (Control Plane) API Specification Update Pull Request 

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

**Comment by audunn** [comment]
On linter differ isues ÞI think there is a bug in the linter. 
It is acceptable for response to POST to be a 200/204 NoContent if the action does not require a response.
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#call-action-post-asynchronously

**Comment by audunn** [comment]
BackupVaults including Backups_GetLatestStatus that is now flagged was armapproved here https://github.com/Azure/azure-rest-api-specs-pr/pull/8331

**Comment by ms-zhenhua** [comment]
Please reach out to Azure Breaking Changes Reviewers azbreakchangereview@microsoft.com to get a breaking changes approval and ARM can review the changes afterwards. Once you have the approval, please remove the `ARMChangesRequested` label from the PR to make the PR visible to the ARM reviewers.

**Comment by audunn** [comment]
> Please reach out to Azure Breaking Changes Reviewers [azbreakchangereview@microsoft.com](mailto:azbreakchangereview@microsoft.com) to get a breaking changes approval and ARM can review the changes afterwards. Once you have the approval, please remove the `ARMChangesRequested` label from the PR to make the PR visible to the ARM reviewers.

@ms-zhenhua  Breaking change was approved here https://msazure.visualstudio.com/One/_workitems/edit/15313053/

**Comment by jiteshmani** [comment]
@audunn Any updates on this, when will this get merged?

**Comment by ms-zhenhua** [comment]
> > Please reach out to Azure Breaking Changes Reviewers [azbreakchangereview@microsoft.com](mailto:azbreakchangereview@microsoft.com) to get a breaking changes approval and ARM can review the changes afterwards. Once you have the approval, please remove the `ARMChangesRequested` label from the PR to make the PR visible to the ARM reviewers.
> 
> @ms-zhenhua Breaking change was approved here https://msazure.visualstudio.com/One/_workitems/edit/15313053/

Hi @audunn, LGTM.

Please get the `Approved-Breakingchange` label from `Azure Breaking Changes Reviewers`. 
Please follow https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process to suppress [GetCollectionOnlyHasValueAndNextLink](https://github.com/Azure/azure-openapi-validator/blob/main/docs/get-collection-only-has-value-and-next-link.md) in https://github.com/Azure/azure-rest-api-specs/pull/24741/checks?check_run_id=15132474961.
After that, it can be signed off.

**Comment by audunn** [comment]
BTW GetCollectionOnlyHasValueAndNextLink link returns a 404.

**Comment by mikekistler** [comment]
Breaking changes previously reviewed and approved in Azure/azure-rest-api-specs-pr#8331

**Comment by audunn** [comment]
@rkmanda @ArcturusZhang Pipiline seems stuck does not run, do you have any idea what may be wrong?

**Comment by TimLovellSmith** [comment]
>       ],

so someone can legitimately send a PATCH request with no properties, and there's nothing else in the payload?

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/preview/2022-11-01-preview/netapp.json:7477 in 801f0ea. [](commit_id = 801f0eaf4a91aa9b81b508a38f0c1bbf50a627ae, deletion_comment = True)

**Comment by TimLovellSmith** [comment]
>       ],

Or did you just have to remove properties because of flattening meaning its not actually even a thing in the payload?

---
In reply to: [1679507476](https://github.com/Azure/azure-rest-api-specs/pull/24741#issuecomment-1679507476) [](http://example.com/codeflow?ancestors=1679507476)

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/preview/2022-11-01-preview/netapp.json:7477 in 801f0ea. [](commit_id = 801f0eaf4a91aa9b81b508a38f0c1bbf50a627ae, deletion_comment = True)

**Review by TimLovellSmith — APPROVED** [review]
:shipit:

**Comment by audunn** [comment]
@rkmanda are we good to have this merged ?

File: specification/netapp/resource-manager/readme.md
Comment (rkmanda) [inline specification/netapp/resource-manager/readme.md:358]: >GetCollectionOnlyHasValueAndNextLink [](http://example.com/codeflow?start=10&length=36)

The explanation for this suppression is unclear. pl elaborate

File: specification/netapp/resource-manager/readme.md
Comment (rkmanda) [inline specification/netapp/resource-manager/readme.md:351]: >LroPostReturn [](http://example.com/codeflow?start=10&length=13)

This is a known issue with the linter rule so this can be approved

File: specification/netapp/resource-manager/readme.md
Comment (audunn) [reply specification/netapp/resource-manager/readme.md:358]: Hi,  
This API was approved in a PR in the private repository in https://github.com/Azure/azure-rest-api-specs-pr/pull/8331 
There where dealys in getting it out but we are now bringing it now in prevew.
Fixing this response model according to the rule would requre changing the reponse which is something the team is trying to avoid this late in the game.

Comment (audunn) [reply specification/netapp/resource-manager/readme.md]: @rkmanda  Would modeling this omething like  `/latestBackupStatus/current` suffice ?  
Will we need the collection GET as well ?

Comment (audunn) [reply specification/netapp/resource-manager/readme.md]: Removed suppression added default name, `GetCollectionOnlyHasValueAndNextLink` does not show.

Comment (audunn) [reply specification/netapp/resource-manager/readme.md]: Another option is to remove this from this version and introduce correctly modeled in the next. The preview clients have a workaround for this.  This would give the team the time to patch the service side code needed.

**Review by ms-henglu — APPROVED** [review]

**Pull request merged**
